pennylane/workflow/construct_batch.py

```diff
@@ -81,8 +83,111 @@ def _get_full_transform_program(
     return program + qnode.device.preprocess_transforms(config)
 
 
+def _validate_level(
+    level: Optional[Literal["top", "user", "device", "gradient"] | int | slice],
+) -> None:
+    """Check that the level specification is valid.
+
+    Args:
+        level: The level specification from user input
+
+    Raises:
+        ValueError: If the level is not recognized
+    """
+    if level is None or isinstance(level, int):
+        return
+
+    if isinstance(level, str):
+        if level not in ("top", "user", "device", "gradient"):
+            raise ValueError(
+                f"level {level} not recognized. Acceptable strings are 'device', 'top', 'user', and 'gradient'."
+            )
+        return
+
+    if isinstance(level, slice):
+        return
```

Comment:
```suggestion
    if level is None or isinstance(level, int) or isinstance(level, slice):
        return

    if isinstance(level, str):
        if level not in ("top", "user", "device", "gradient"):
            raise ValueError(
                f"level {level} not recognized. Acceptable strings are 'device', 'top', 'user', and 'gradient'."
            )
        return
```

Might as well combine the conditions.